File: midterm/src/app/services/fetch/order.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { ConfirmedOrderRequest, OrderResponse } from '../../models/order';
import { Paging } from '../../models/paging';

@Injectable({
  providedIn: 'root'
})
export class OrderService {
  constructor(private readonly baseService: BaseService) { }

  getAllInCart(): Observable<OrderResponse[]> {
    return this.baseService.get<OrderResponse[]>('/api/orders/cart');
  }

  getHistory(page: number): Observable<Paging<OrderResponse>> {
    return this.baseService.get<Paging<OrderResponse>>(`/api/orders/history`, { page });
  }

  addToCart(productId: number): Observable<any> {
    return this.baseService.post<OrderResponse>('/api/orders/cart', { productId });
  }

  confirmOrder(order: ConfirmedOrderRequest): Observable<any> {
    return this.baseService.post<OrderResponse>('/api/orders/', order);
  }

  removeFromCart(productId: number): Observable<any> {
    return this.baseService.delete<OrderResponse>(`/api/orders/${productId}/`);
  }
}

File: java-midterm/src/main/java/com/pht/service/OrderService.java
package com.pht.service;

import java.util.List;

import com.pht.dto.CustomPaging;
import com.pht.dto.request.ConfirmedOrderRequest;
import com.pht.dto.request.OrderRequest;
import com.pht.dto.response.OrderResponse;

public interface OrderService {
	List<OrderResponse> getAllInCart();
	
	CustomPaging<OrderResponse> getOrderHistory(int page);
	
	void confirmedOrder(ConfirmedOrderRequest request);
	
	void addToCart(OrderRequest request);
	
	void delete(long id);
}
